harvester_e2e_tests/integration/test_vm_snapshot.py

```diff
@@ -0,0 +1,609 @@
+
+    userdata = yaml.safe_load(vm_spec.user_data)
+    userdata["ssh_authorized_keys"] = [pubkey]
+    vm_spec.user_data = yaml.dump(userdata)
+
+    _, data = api_client.vms.create(name, vm_spec)
+    deadline = datetime.now() + timedelta(seconds=timeout_secs)
+    while deadline > datetime.now():
+        _, data = api_client.vms.get(name)
+        if "Running" == data.get("status", {}).get("printableStatus"):
+            break
+        sleep(1)
+    else:
+        raise AssertionError(f"timed out waiting for {name} to transition to Running")
+
+    return name, image["user"]
+
+
+def delete_vm(name, api_client, timeout_secs):
+    code, data = api_client.vms.get(name)
+    if code == 404:
+        return
+
+    vm_spec = api_client.vms.Spec.from_dict(data)
+
+    api_client.vms.delete(name)
+    deadline = datetime.now() + timedelta(seconds=timeout_secs)
+    while deadline > datetime.now():
+        code, data = api_client.vms.get_status(name)
+        if 404 == code:
+            break
+        sleep(3)
+
+    for vol in vm_spec.volumes:
+        vol_name = vol["volume"]["persistentVolumeClaim"]["claimName"]
+        api_client.volumes.delete(vol_name)
+
+
+def stop_vm(name, api_client, timeout_secs):
+    _, data = api_client.vms.stop(name)
+    deadline = datetime.now() + timedelta(seconds=timeout_secs)
+    while deadline > datetime.now():
+        _, data = api_client.vms.get(name)
+        status = data.get("status", {}).get("printableStatus")
+        if "Stopped" == status:
+            return
+        sleep(1)
+
+    raise AssertionError(f"timed out trying to stop {name}")
+
+
+@pytest.fixture(scope="class")
+def source_vm(sourcevm_name, api_client, ssh_keypair, image, wait_timeout):
+    yield create_vm(sourcevm_name, api_client, ssh_keypair, image, wait_timeout)
+    delete_vm(sourcevm_name, api_client, wait_timeout)
+
+
+@pytest.fixture(scope="class")
+def restored_from_snapshot_vm(api_client, restored_from_snapshot_name, vm_snapshot, wait_timeout):
+    spec = api_client.vm_snapshots.RestoreSpec.for_new(restored_from_snapshot_name)
+    code, data = api_client.vm_snapshots.restore(vm_snapshot, spec)
+    assert 201 == code
+
+    deadline = datetime.now() + timedelta(seconds=wait_timeout)
+    while deadline > datetime.now():
+        code, data = api_client.vms.get(restored_from_snapshot_name)
+        if 200 == code and "Running" == data.get("status", {}).get("printableStatus"):
+            break
+        print("waiting for restored vm to be running")
+        sleep(3)
+
+    yield restored_from_snapshot_name
+
+    delete_vm(restored_from_snapshot_name, api_client, wait_timeout)
+
+
+@pytest.fixture(scope="class")
+def backup_source_vm(newvm_name, api_client, ssh_keypair, image, wait_timeout):
+    yield create_vm(newvm_name, api_client, ssh_keypair, image, wait_timeout)
+    delete_vm(newvm_name, api_client, wait_timeout)
+
+
+@pytest.fixture(scope="class")
+def vm_backup(vm_backup_name, backup_source_vm,
```

Comment:
Are these backup related fixtures required?

Reply:
No, I've removed them from the latest revision.